### PR TITLE
Updated the usage description code snippet so that it actually ignore…

### PR DESCRIPTION
**Pull request by dlochrie**
…s the error (as demonstated [here](http://stackoverflow.com/a/21866122/1058612)). Also made some minor grammar fixes, and updated docs so that they fit within the 80 character margin (as most of the doc does).

### CODE REVIEW

File: doc.go [hunk 1]
```diff
@@ -29,13 +29,9 @@ Let's start with an example that shows the sessions API in a nutshell:
...
 	func MyHandler(w http.ResponseWriter, r *http.Request) {
-		// Get a session. We're ignoring the error resulted from decoding an
+		// Get a session. We're ignoring any error that occurs while decoding an
 		// existing session: Get() always returns a session, even if empty.
-		session, err := store.Get(r, "session-name")
-		if err != nil {
-			http.Error(w, err.Error(), 500)
-			return
-		}
+		session, _ := store.Get(r, "session-name")
```
Comment (elithrar) [inline doc.go]: This should be the other way around: we should always check the error in example code and remove the 'we're ignoring...' text from the comment.

File: doc.go [hunk 2]
```diff
@@ -117,14 +113,16 @@ so it is easy to register new datatypes for storage in sessions:
...
 
-As it's not possible to pass a raw type as a parameter to a function, gob.Register()
-relies on us passing it an empty pointer to the type as a parameter. In the example
-above we've passed it a pointer to a struct and a pointer to a custom type
-representing a map[string]interface. This will then allow us to serialise/deserialise
-values of those types to and from our sessions.
+As it's not possible to pass a raw type as a parameter to a function,
+gob.Register() relies on us passing it an empty pointer to the type as a
+parameter. In the example above we've passed it a pointer to a struct and a
+pointer to a custom type representing a map[string]interface. This will then
```
Comment (elithrar) [inline doc.go]: Might as well fix it to say `map[string]interface{}` while we're here.

**Comment by elithrar** [comment]
Thanks for the PR. Have added a couple of comments and have updated my SO answer (the one you linked to) to explicitly check the error.

Example code should always check for errors as we need to set a good example of newcomers to the language.

File: doc.go (hunk 1 above)
Comment (dlochrie) [inline doc.go]: That makes sense. The text has been reverted/updated.

File: doc.go (hunk 2 above)
Comment (dlochrie) [inline doc.go]: Done.

**Comment by dlochrie** [comment]
Updated per comments.

**Comment by elithrar** [comment]
Thanks again for this @dlochrie — forgot I hadn't merged this!

**Pull request closed without merging**